server/tests/src/test/java/org/infinispan/server/cli/CliIT.java

```diff
@@ -108,7 +108,15 @@ public void testCliInteractive() {
          terminal.clear();
          terminal.send("query \"from org.infinispan.rest.search.entity.Person p where p.gender = 'MALE'\"");
          terminal.assertContains("\"total_results\":3,");
+         terminal.clear();
 
+         terminal.send("index stats qcache");
+         terminal.assertContains("\"slowest\" : \"from org.infinispan.rest.search.entity.Person p where p.gender = 'MALE'\"");
+         terminal.clear();
+         terminal.send("index clear qcache");
+         terminal.send("index reindex qcache");
+         terminal.send("index clear-stats qcache");
```

Comment:
Maybe we can test that after the clear there are no more match and after the reindex we will have some matches again.

Reply:
unfortunately those are async calls which don't wait for completion. We could introduce an artificial delay, or probably poll the stats to check completion of the task.